### PR TITLE
Notification settings functionality

### PR DESCRIPTION
**Pull request by hedgehog1833**
Finally fetching and updating all notification functionality works fine 👌 
I had to downgrade the styled-components Lib because of compilation errors.

### CODE REVIEW

File: src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  const handleNotifyReissues = (event: React.SyntheticEvent, checked: boolean) => {
+    event.preventDefault()
+    handleUpdate({
+      frequencyInWeeks: frequencyInWeeks,
+      notificationAtReleaseDate: notificationAtReleaseDate,
+      notificationAtAnnouncementDate: notificationAtReleaseDate,
+      notifyReissues: checked,
+    })
+  }
```
Comment (DanielW1987) [inline src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx:92]: I don't think that scales very well in general. 

Is there a reason why you didn't implement this with useEffect?

Something like this:

```typescript
  useEffect(() => {
    handleUpdate({
      frequencyInWeeks: frequencyInWeeks,
      notificationAtReleaseDate: notificationAtReleaseDate,
      notificationAtAnnouncementDate: notificationAtAnnouncementDate,
      notifyReissues: notifyReissues,
    })
  }, [frequencyInWeeks, notificationAtReleaseDate, notificationAtAnnouncementDate, notifyReissues])
```

File: src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        channel: props.channel,
+        ...newNotificationConfig!,
+      },
+      {
+        onSuccess: () => {
+          setFrequencyInWeeks(newNotificationConfig!.frequencyInWeeks)
+          setNotificationAtReleaseDate(newNotificationConfig!.notificationAtReleaseDate)
+          setNotificationAtAnnouncementDate(newNotificationConfig!.notificationAtAnnouncementDate)
+          setNotifyReissues(newNotificationConfig!.notifyReissues)
+        },
```
Comment (DanielW1987) [inline src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx:46]: I think you don't need that if you have a proper state binding (see my other comment)

```suggestion
```

File: src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  const { mutate: updateNotificationConfig } = useUpdateNotificationConfig()
+
+  useEffect(() => {
+    if (props.notificationConfig) {
+      setFrequencyInWeeks(props.notificationConfig.frequencyInWeeks)
+      setNotificationAtReleaseDate(props.notificationConfig.notificationAtReleaseDate)
+      setNotificationAtAnnouncementDate(props.notificationConfig.notificationAtAnnouncementDate)
+      setNotifyReissues(props.notificationConfig.notifyReissues)
+    }
+  }, [props.notificationConfig])
```
Comment (DanielW1987) [inline src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx:32]: I think you don't need that if you have a proper state binding.

e.g.

```typescript
<FormControl>
  <FormControlLabel
      control={<Switch color={'info'} value={notifyReissues} />}
      label={'Notifications for reissues or re-releases'}
      onChange={(_, checked) => setNotifyReissues(checked)}
  />
</FormControl>
```

```suggestion
```

**Review by DanielW1987 — APPROVED** [review]
There is some potential for refactoring. Fine to do it with a follow-up PR.

File: src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx (hunk 1 above)
Comment (hedgehog1833) [reply src/Components/NotificationConfig/DefaultNotificationConfigOptions.tsx:92]: For some reason I just don't get this running in any other way 🤔 Just tried again for half an hour 😅 same with your other comments. I'll merge this and would be happy if you would give it a try sometime :)

**Pull request merged**